library/src/scala/internal/quoted/Matcher.scala

```diff
@@ -246,36 +270,36 @@ object Matcher {
       }
     }
 
-    def treeOptMatches(scrutinee: Option[Tree], pattern: Option[Tree]) given Context, Env: Matching = {
+    private def treeOptMatches(scrutinee: Option[Tree], pattern: Option[Tree]) given Context, Env: Matching = {
       (scrutinee, pattern) match {
         case (Some(x), Some(y)) => x =#= y
         case (None, None) => matched
         case _ => notMatched
       }
     }
 
-    def caseMatches(scrutinee: CaseDef, pattern: CaseDef) given Context, Env: Matching = {
+    private def caseMatches(scrutinee: CaseDef, pattern: CaseDef) given Context, Env: Matching = {
       val (caseEnv, patternMatch) = scrutinee.pattern =%= pattern.pattern
       withEnv(caseEnv) {
         patternMatch &&
-        treeOptMatches(scrutinee.guard, pattern.guard) &&
-        scrutinee.rhs =#= pattern.rhs
+          treeOptMatches(scrutinee.guard, pattern.guard) &&
+          scrutinee.rhs =#= pattern.rhs
       }
     }
 
     /** Check that the pattern trees match and return the contents from the pattern holes.
      *  Return a tuple with the new environment containing the bindings defined in this pattern and a matching.
      *  The matching is None if the pattern trees do not match otherwise return Some of a tuple containing all the contents in the holes.
      *
-     * @param scrutinee The pattern tree beeing matched
-     * @param pattern The pattern tree that the scrutinee should match. Contains `patternHole` holes.
-     * @param `the[Env]` Set of tuples containing pairs of symbols (s, p) where s defines a symbol in `scrutinee` which corresponds to symbol p in `pattern`.
-     * @return The new environment containing the bindings defined in this pattern tuppled with
-     *         `None` if it did not match or `Some(tup: Tuple)` if it matched where `tup` contains the contents of the holes.
+     *  @param scrutinee The pattern tree beeing matched
+     *  @param pattern The pattern tree that the scrutinee should match. Contains `patternHole` holes.
+     *  @param `the[Env]` Set of tuples containing pairs of symbols (s, p) where s defines a symbol in `scrutinee` which corresponds to symbol p in `pattern`.
+     *  @return The new environment containing the bindings defined in this pattern tuppled with
+     *          `None` if it did not match or `Some(tup: Tuple)` if it matched where `tup` contains the contents of the holes.
      */
-    def (scrutinee: Pattern) =%= (pattern: Pattern) given Context, Env: (Env, Matching) = (scrutinee, pattern) match {
```

Comment:
I'd like to find out more about the rationale behind the symbols `=#=`, `=%=` and similar. Is this written up somewhere?

Reply:
The only description is in the scaladoc of those methods.

Mainly I added to make a small DSL for pattern matching that looks like what would become it's spec. `=#=` and `=%=` stand for "do the trees match? And return the result." and `&&` is a shortcutting `AND` that accumulated the results.

For example
```scala
        case (If(cond1, thenp1, elsep1), If(cond2, thenp2, elsep2)) =>
          cond1 =#= cond2 && thenp1 =#= thenp2 && elsep1 =#= elsep2
```
the second line will check that the condition and two branches match. If so it will aslo accumulate all the pattern splices it found in each on the checks.

Reply:
OK. But that still leaves questions open:

 - Why have both `=#=` and `=%=`. What's the difference between them?
 - I believe the choice of `=#=` is very confusing since `#` means NOT EQUAL in some languages. I'd prefer `=?=`. Or, probably much clearer, use a mixed operator like `pattern_==`.

Is that a user facing API? If yes, I would advise against inventing new symbolic operators.

Reply:
This is only an internal interface. I will change it later to `=?=`.

I had to use different symbols due to limitations of extension methods at the time, but this limitation has been lifted.

Reply:
OK, great. Let's use `=?=` for both of them then.

Reply:
I updated it to use `=?=` instead of `=#=`, `=##=` and `=%=`.